draft-ietf-tls-esni.md

```diff
@@ -180,68 +184,90 @@ ClientHelloInner and either terminate the connection (in Shared Mode)
 or forward it to the backend server (in Split Mode).
 
 
-# Encrypted SNI Configuration {#esni-configuration}
+# Encrypted SNI Configuration {#echo-configuration}
 
 SNI Encryption configuration information is conveyed with the following
-ESNIConfig structure.
+ECHOConfigs structure.
 
 ~~~~
-    // Copied from TLS 1.3
-    struct {
-        NamedGroup group;
-        opaque key_exchange<1..2^16-1>;
-    } KeyShareEntry;
+    opaque HPKEPublicKey<1..2^16-1>;
+    uint16 HPKEKEMID; // Defined in I-D.irtf-cfrg-hpke
 
     struct {
         uint16 version;
+        opaque contents<1..2^16-1>;
+    } ECHOConfig;
+
+    struct {
         opaque public_name<1..2^16-1>;
-        KeyShareEntry keys<4..2^16-1>;
-        CipherSuite cipher_suites<2..2^16-2>;
+
+        HPKEPublicKey public_key;
+        HPKEKEMID kem_id;
+        Ciphersuite cipher_suites<2..2^16-2>;
+
         uint16 maximum_name_length;
         Extension extensions<0..2^16-1>;
-    } ESNIConfig;
+    } ECHOConfigContents;
```

Comment:
I don't understand this. Is the idea that EchoConfig.contents == ECHOConfigContents?

The usual way to denote this is to have at embedded instead.

```
struct {
   uint16 version;
   uint16 length;
   EchoConfigContents....
}
```

Reply:
That works, too!

Reply:
Should be fixed now!